### PR TITLE
PanelInspector: Adds a Raw display mode but defaults to Formatted display mode 

### LINKED ISSUE #25130: Secondary time columns not formatted in CSV export
**What happened**:
When exporting CSV from PanelInspect in 7.0.1, the primary time column is formatted according to the column's formatting preferences as expected (the release notes reference #24992). Any additional time columns however are not formatted. In our table with multiple time columns the secondary times are appearing in ISO8601 format which our business users are unable to deal with.

**What you expected to happen**:
All time columns formatted according to their formatting preferences.

**How to reproduce it (as minimally and precisely as possible)**:
Export a table with multiple time columns to CSV.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.0.1
- Data source type & version: PostgreSQL 10
- OS Grafana is installed on: Oracle Linux 6.10
- User OS & Browser: Windows 10, Chrome 81
- Grafana plugins:
- Others:

### LINKED ISSUE #26479: Grafana 6.6.2 -> 7.10 - Exporting table with value mappings now exports the key, not the mapped value
**What happened**:
Tried to export table visualisation. The query returns arbitrary integers which I map to Text using Grafana. Usually when exporting, the mapped values are exported. However, since upgrading to 7.1.0, it now exports the integers instead of the mapped values

**What you expected to happen**:
Exported data would be of the value mappings.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create Table visualisation, return some kind of result with random integers in it. I use Clickhouse so used: select * from system.numbers limit 5.
2. this query returns me 5 rows of 0,1,2,3,4
3. Set up a value mapping of 0 -> "test".
4. Try to export Data, and you'll get 0,1,2,3,4

**Anything else we need to know?**:

Occurred when upgrading from 6.6.2 to 7.1.0

**Environment**:
- Grafana version: 7.10
- Data source type & version: N/A
- OS Grafana is installed on: Ubuntu 18.04
- User OS & Browser: Windows 10 Chrome, Firefox
- Grafana plugins: Standard Table
- Others:

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Previously fields with overrides stored their raw values in the CSV instead of the chosen field overrides. A customer reported this issue.

A cleaned up code in PanelInspector that I don't think we need anymore.

**Which issue(s) this PR fixes**:
Fixes #26479
Fixes #25130

**Special notes for your reviewer**:
I'm out on deep water here as I'm not familiar with all the displayProcessor and CSV gotchas at all. Please let me know if I'm way off.

### CODE REVIEW

File: packages/grafana-data/src/utils/csv.ts [hunk 1]
```diff
@@ -205,6 +206,13 @@ function writeValue(value: any, config: CSVConfig): string {
 }
 
 function makeFieldWriter(field: Field, config: CSVConfig): FieldWriter {
+  if (field.display) {
+    return (value: any) => {
+      const displayValue = field.display!(value);
+      return writeValue(formattedValueToString(displayValue), config);
+    };
+  }
+
```
Comment (hugohaggmark) [inline packages/grafana-data/src/utils/csv.ts:215]: The if statements below for `boolean` and `number` could probably be removed too? WDYT?

File: public/app/features/dashboard/components/Inspector/InspectDataTab.tsx [hunk 2]
```diff
@@ -60,33 +55,6 @@ export class InspectDataTab extends PureComponent<Props, State> {
...
-
-      const fields = [...dataFrame.fields];
-      fields[timeIndex!] = formattedDateField;
-
-      dataFrame = {
-        ...dataFrame,
-        fields,
-      };
-    }
-
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/Inspector/InspectDataTab.tsx:89]: Couldn't see why this was necessary anymore.

File: packages/grafana-data/src/utils/csv.ts (hunk 1 above)
Comment (hugohaggmark) [reply packages/grafana-data/src/utils/csv.ts:215]: On second thought, they should stay when we don't use field overrides setting CSV.

**Comment by torkelo** [comment]
This does have the the impact I expected, I have to check "apply field config"  to get the raw values with all the decimals. Which is not true in all cases, if I had field config set in field config to 1 decimal I would have no way to download csv with the original raw decimal precision. 

Not sure how to fix this and what the default behavior should be for the table & csv export

**Comment by hugohaggmark** [comment]
> This does have the the impact I expected, I have to check "apply field config" to get the raw values with all the decimals. Which is not true in all cases, if I had field config set in field config to 1 decimal I would have no way to download csv with the original raw decimal precision.
> 
> Not sure how to fix this and what the default behavior should be for the table & csv export

Not sure I follow, if you uncheck `apply field config` doesn't it show the raw value then, I mean the value without field override?

**Comment by torkelo** [comment]
no, if I have a value like 23.1231231  it gets rendered as 23  in the csv (and data table) and a value like `"23.232323"` in quotes get the same results, have to check the apply field override to get the decimals to show

**Comment by hugohaggmark** [comment]
> no, if I have a value like 23.1231231 it gets rendered as 23 in the csv (and inspect table) and a value like `"23.232323"` in quotes get the same results, have to check the apply field override to get the decimals to show

I think I understand now, thanks. Then I guess I'll close this PR as this sounds like a bigger issue that we need to address?

**Comment by torkelo** [comment]
the problem is that we call applyFieldOverrides even when apply field config is not checked, https://github.com/grafana/grafana/blob/b65923923f087215b4a92911aefbdc998996be84/public/app/features/dashboard/components/Inspector/InspectDataTab.tsx#L119  

which does apply display processors. This is super tricky, maybe if apply field config is not checked we have csv not use display processor? But before addressing that I think we need to think a bit on what the table is showing, should the CSV export match what we show in the table? If yes, we need to apply consistent value rendering. 

Tricky thing is date's really need formatting to make sense of them (epochs), while values don't

**Review by mckn — APPROVED** [review]
LGTM

Looks really good and works as expected when I took it for a spin! Great job!

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 3]
```diff
@@ -427,3 +428,35 @@ export const getLinksSupplier = (
...
+
+  for (let frameIndex = 0; frameIndex < newData.length; frameIndex++) {
+    const newFrame = { ...newData[frameIndex] };
+    const newFields = [...newFrame.fields];
+
+    for (let fieldIndex = 0; fieldIndex < newFields.length; fieldIndex++) {
+      newFields[fieldIndex] = {
+        ...newFields[fieldIndex],
+        display: processor,
+        values: new FormattedVector(newFields[fieldIndex].values, processor),
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts:451]: This should not be needed right? one display processor on the field should be enough

**Review by torkelo — APPROVED** [review]
Looking good

File: packages/grafana-data/src/field/fieldOverrides.ts (hunk 3 above)
Comment (hugohaggmark) [reply packages/grafana-data/src/field/fieldOverrides.ts:451]: I'll look into it

Comment (hugohaggmark) [reply packages/grafana-data/src/field/fieldOverrides.ts:451]: Yes, that works. I copy/pasted the old code. Will be awesome to have more documentation about best practices on how to create/update/copy/delete data frames. (ping @marcusolsson)

**Pull request merged**